### ISSUE TITLE
[a11y][Violation][ARIA specification] Accessibility issues reported on JupyterLab UI using IBM Equal Access Accessibility Checker tool

### ISSUE THREAD

**Issue by nkn2022** [jupyterlab/jupyterlab#16891]
## Description

I am running an accessibility test suite with Jupyter using the **IBM Equal Access Accessibility Checker** tool on Chrome browser. The results look good overall, only a few things are missing to reach a first formal level of compliance. So kindly help look into these. 

Issues reported were violating **WCAG 2.2** requirements. 
Ref: https://www.ibm.com/able/requirements/checker-rule-sets

Listing of violations categorized based on the checkpoints. 
Full report (includes information on element locations where issue were found): 
[Accessibility_Report-Untitled.ipynb - JupyterLab.xlsx](https://github.com/user-attachments/files/17383160/Accessibility_Report-Untitled.ipynb.-.JupyterLab.xlsx)

# Category: `ARIA specification`
Ref: 

****Violation of Rule:** `aria_attribute_valid`**

> List of issues with elements that violate this rule with a screenshots
> > **Issue 1** 
> > The ARIA attributes "aria-pressed" are not valid for the element `<jp-button>` with ARIA role "none"
> > **Element location:** 
> > `<jp-button minimal="" appearance="stealth" current-value="" tabindex="-1" title="Upload Files" aria-pressed="false" aria-label="Upload Files" aria-disabled="false" class="jp-ToolbarButtonComponent">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/8a0c1aa2-a9ba-4753-81e3-d15ea73d0a35">

> > **Issue 2**
> > The ARIA attributes "aria-pressed" are not valid for the element `<jp-button>` with ARIA role "none"
> > **Element location:** 
> > `<jp-button minimal="" appearance="stealth" current-value="" tabindex="-1" title="Enable Debugger" aria-pressed="false" aria-label="Enable Debugger" aria-disabled="false" class="jp-DebuggerBugButton jp-ToolbarButtonComponent">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/0c226e11-1236-411b-b4f4-23761d6a1f1b">

> > **Issue 3**
> > The ARIA attributes "aria-label" are not valid for the element `<div>` with implicit ARIA role "generic"
> > **Element location:** 
> > `<div data-windowed-list-index="0" tabindex="0" aria-label="Code Cell Content with Output" class="lm-Widget jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-selected jp-mod-active">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/c8b1261a-2de7-49f9-a03e-b1cbff7829ae">

> > **Issue 4**
> > The ARIA attributes "aria-label" are not valid for the element `<div>` with implicit ARIA role "generic"
> > **Element location:** 
> > `<div data-windowed-list-index="1" tabindex="-1" aria-label="Code Cell Content with Output" class="lm-Widget jp-Cell jp-CodeCell jp-Notebook-cell">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/31aafec7-8258-4fec-83e7-13e0bdf4d7b5">

> > **Issue 5**
> > The ARIA attributes "aria-label" are not valid for the element `<div>` with implicit ARIA role "generic"
> > **Element location:** 
> > `<div data-windowed-list-index="2" tabindex="-1" aria-label="Code Cell Content" class="lm-Widget jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-noOutputs">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/db864134-bc3b-46ce-b278-71207cf94c93">

## Reproduce

- Install Chrome extension for IBM accessibility checker tool. Ref: https://www.ibm.com/able/toolkit/verify/automated
- Open JupyterLab UI on Chrome browser. Open `notebook` file to edit.
- Run the tool and we can see the above `Violation`s reported.

Here is the screenshot of the UI scanned for the attached report:

<img width="1534" alt="image" src="https://github.com/user-attachments/assets/0e0cbc5c-cf21-4393-97fe-b566da24b58b">

## Expected behavior

No violations are reported on the scanning to reach a first formal level of compliance.

## Context

- Operating System and version: MacOS Sonoma Version 14.5
- Browser and version: Chrome Version 129+
- JupyterLab version: Version 4.2.5

**Comment by nkn2022** [comment]
Re-tested on Notebook v7.4.7. All issues still exists. The description has been updated to reflect the latest scans. 

****Violation of Rule:** `aria_attribute_valid`** 
> > **Issue 1** 
> > **Issue 2**
